operator/internal/manifests/config.go

```diff
@@ -132,6 +132,11 @@ func ConfigOptions(opt Options) config.Options {
 		}
 	}
 
+	if opt.ObjectStorage.S3 != nil {
+		awsEndpointSuffix := ".amazonaws.com"
+		opt.ObjectStorage.S3.ForcePathStyle = !strings.HasSuffix(opt.ObjectStorage.S3.Endpoint, awsEndpointSuffix)
+	}
```

Comment:
I suggested to put actually the flag setting into [here](https://github.com/grafana/loki/blob/main/operator/internal/handlers/internal/storage/secrets.go#L391-L461) instead of the `ConfigOptions` func because you mutate an input here. Best place for visibility of such mutation is the place where we originaly construct the object storage options, i.e. when we extract from secrets.

Reply:
Oh I remember now that we agreed on that, sorry!